zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/listeners/task/TaskListenerTest.java

```diff
@@ -1057,6 +1057,67 @@ public void shouldAppendUserTaskCorrectedWhenTaskListenerCompletesWithCorrection
         .hasUserTaskKey(userTask.getUserTaskKey());
   }
 
+  @Test
+  public void shouldProvideCorrectedUserTaskDataToSubsequentTaskListener() {
+    // given
+    final long processInstanceKey =
+        createProcessInstance(
+            createProcessWithCompleteTaskListeners(LISTENER_TYPE, LISTENER_TYPE + "_2"));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).complete();
+
+    // when
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(LISTENER_TYPE)
+        .withResult(
+            new JobResult()
+                .setCorrections(
+                    new JobResultCorrections()
+                        .setAssignee("new_assignee")
+                        .setCandidateUsers(List.of("new_candidate_user"))
+                        .setCandidateGroups(List.of("new_candidate_group"))
+                        .setDueDate("new_due_date")
+                        .setFollowUpDate("new_follow_up_date")
+                        .setPriority(100))
+                .setCorrectedAttributes(
+                    List.of(
+                        "assignee",
+                        "candidateUsers",
+                        "candidateGroups",
+                        "dueDate",
+                        "followUpDate",
+                        "priority")))
+        .complete();
+
+    // then
+    final var activatedListenerJob = activateJob(processInstanceKey, LISTENER_TYPE + "_2");
+    assertThat(activatedListenerJob.getCustomHeaders())
+        .describedAs("Expect that corrected data is accessible in the subsequent listener")
+        .contains(
+            entry(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME, "new_assignee"),
+            entry(Protocol.USER_TASK_CANDIDATE_USERS_HEADER_NAME, "[\"new_candidate_user\"]"),
+            entry(Protocol.USER_TASK_CANDIDATE_GROUPS_HEADER_NAME, "[\"new_candidate_group\"]"),
+            entry(Protocol.USER_TASK_DUE_DATE_HEADER_NAME, "new_due_date"),
+            entry(Protocol.USER_TASK_FOLLOW_UP_DATE_HEADER_NAME, "new_follow_up_date")
+            /*
+             // priority is not yet accessible as a custom header
+             , entry(Protocol.USER_TASK_PRIORITY_HEADER_NAME, "100")
+            */
```

Comment:
👍🏻 Thanks, Nico! 🙌
I appreciate you adding this hint about the `priority` header. I'll uncomment it as part of the [Provide access to `action` and `priority` user task properties to task listeners](https://github.com/camunda/camunda/issues/24102) task.